src/main/java/org/prnhs/javaee/swim/services/ProgramService.java

```diff
@@ -6,40 +6,40 @@
 import org.prnhs.javaee.swim.core.dao.ProgramDao;
 import org.prnhs.javaee.swim.core.entity.Program;
 import org.prnhs.javaee.swim.dto.ProgramDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
 public class ProgramService {
     
+    private static final Logger LOGGER = LoggerFactory.getLogger(ProgramService.class);
+    
     @Autowired
     private ProgramDao dao;
     
     public ProgramDto save(ProgramDto programDto){
-        if(programDto == null){
+        if(programDto.getObjective() == null){
+            LOGGER.debug("Save method called in ProgramService but the program object was null");
             throw new IllegalArgumentException("hey, you were supposed to give me a program!");
         }
-        Program program = dao.findOne(programDto.getId());
-        
-        if(program == null){
-            program = ProgramTranslator.toEntity(programDto);
-        }
-        else{
-            program.setObjective(programDto.getObjective());
-        }
         
+        Program program = ProgramTranslator.toEntity(programDto);        
         program = dao.save(program);
         
         ProgramDto savedProgram = ProgramTranslator.toDto(program);
-        
+        LOGGER.debug("Save method executed successfully in ProgramService, the program with id: {} got saved", savedProgram.getId());

```

Comment:
print the entire dto.  Don't forget that you don't need to specify toString() on the object you want to print.  The .debug() method will do that for you if it needs to.